fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/models/FedoraResourceImpl.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.impl.models;
+
+import java.net.URI;
+import java.time.Instant;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.fcrepo.kernel.api.RdfStream;
+import org.fcrepo.kernel.api.Transaction;
+import org.fcrepo.kernel.api.exception.ItemNotFoundException;
+import org.fcrepo.kernel.api.models.FedoraResource;
+import org.fcrepo.kernel.api.models.ResourceHeaders;
+import org.fcrepo.persistence.api.PersistentStorageSessionFactory;
+import org.fcrepo.persistence.api.exceptions.PersistentItemNotFoundException;
+
+/**
+ * Implementation of a Fedora resource, containing functionality common to the more concrete resource implementations.
+ *
+ * @author bbpennel
+ */
+public class FedoraResourceImpl implements FedoraResource {
+    private PersistentStorageSessionFactory pSessionFactory;
```

Comment:
Is this here to lazy load some data?

Reply:
Yes, see the implementations of getTriples, getManagedProperties. Also, in FedoraBinary, getContent would lazy load.
https://github.com/fcrepo4/fcrepo4/pull/1557/files#diff-b3f3902fc824f0ad14ce5c56fe6afca8R208